navit/navit.c

```diff
@@ -3504,6 +3504,27 @@ void navit_layout_switch(struct navit *n) {
                 && valid_attr.u.num==attr_position_valid_invalid) {
             return; //No valid fix yet
         }
+
+        int *streetflags = tracking_get_current_flags(n->tracking);
+
+        if (streetflags != NULL) {
+
+            dbg(lvl_debug,"streetflags: %i", *streetflags);
+
+            // We are in  a tunnel and we have a nightlayout -> switch to nightlayout
+            if ((*streetflags & AF_UNDERGROUND) && l->nightname) {
```

Comment:
Suggest factoring out check for "l->nightname" or check for "AF_UNDERGROUND". Both allow to get rid of the ugly else if.